sumologic/resource_sumologic_kinesis_log_source_test.go

```diff
@@ -0,0 +1,204 @@
+	})
+}
+func testAccCheckKinesisLogSourceDestroy(s *terraform.State) error {
+	client := testAccProvider.Meta().(*Client)
+	for _, rs := range s.RootModule().Resources {
+		if rs.Type != "sumologic_kinesis_log_source" {
+			continue
+		}
+		if rs.Primary.ID == "" {
+			return fmt.Errorf("HTTP Source destruction check: HTTP Source ID is not set")
+		}
+		id, err := strconv.Atoi(rs.Primary.ID)
+		if err != nil {
+			return fmt.Errorf("Encountered an error: " + err.Error())
+		}
+		collectorID, err := strconv.Atoi(rs.Primary.Attributes["collector_id"])
+		if err != nil {
+			return fmt.Errorf("Encountered an error: " + err.Error())
+		}
+		s, err := client.GetKinesisLogSource(collectorID, id)
+		if err != nil {
+			return fmt.Errorf("Encountered an error: " + err.Error())
+		}
+		if s != nil {
+			return fmt.Errorf("KinesisLog Source still exists")
+		}
+	}
+	return nil
+}
+func testAccCheckKinesisLogSourceExists(n string, kinesisLogSource *KinesisLogSource) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		rs, ok := s.RootModule().Resources[n]
+		if !ok {
+			return fmt.Errorf("not found: %s", n)
+		}
+		if rs.Primary.ID == "" {
+			return fmt.Errorf("KinesisLog Source ID is not set")
+		}
+		id, err := strconv.Atoi(rs.Primary.ID)
+		if err != nil {
+			return fmt.Errorf("KinesisLog Source id should be int; got %s", rs.Primary.ID)
+		}
+		collectorID, err := strconv.Atoi(rs.Primary.Attributes["collector_id"])
+		if err != nil {
+			return fmt.Errorf("Encountered an error: " + err.Error())
+		}
+		c := testAccProvider.Meta().(*Client)
+		kinesisLogSourceResp, err := c.GetKinesisLogSource(collectorID, id)
+		if err != nil {
+			return err
+		}
+		*kinesisLogSource = *kinesisLogSourceResp
+		return nil
+	}
+}
+func testAccCheckKinesisLogSourceValues(kinesisLogSource *KinesisLogSource, name, description, category string) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		if kinesisLogSource.Name != name {
+			return fmt.Errorf("bad name, expected \"%s\", got: %#v", name, kinesisLogSource.Name)
+		}
+		if kinesisLogSource.Description != description {
+			return fmt.Errorf("bad description, expected \"%s\", got: %#v", description, kinesisLogSource.Description)
+		}
+		if kinesisLogSource.Category != category {
+			return fmt.Errorf("bad category, expected \"%s\", got: %#v", category, kinesisLogSource.Category)
+		}
+		return nil
+	}
+}
+func testAccSumologicKinesisLogSourceConfig(cName, cDescription, cCategory, sName, sDescription, sCategory, testAwsRoleArn, testAwsBucket string) string {
+	return fmt.Sprintf(`
+resource "sumologic_collector" "test" {
+	name = "%s"
+	description = "%s"
+	category = "%s"
+}
+
+resource "sumologic_kinesis_log_source" "kinesisLog" {
+	name = "%s"
+	description = "%s"
+	category = "%s"
+	content_type  = "KinesisLog"
+	collector_id = "${sumologic_collector.test.id}"
+	authentication {
+		type = "AWSRoleBasedAuthentication"
+		role_arn = "%s"
+	  }
+	path {
+		type = "KinesisLogPath"
```

Comment:
nit: fix indentation

Reply:
indentation is still messed up in multiple places in this file, not sure why the conversation was marked resolved